doc/source/developer.rst

```diff
@@ -106,6 +108,8 @@ As an example of fully-formed metadata:
 .. code-block:: text
 
    {'index_columns': ['__index_level_0__'],
+    'column_index_names': [None],
+    'column_index_dtypes': ['object'],
```

Comment:
We may want to enrich these types to include both the pandas type and the numpy type

Reply:
Yep, I'm going to follow up here.